cmd/init/init.go

```diff
@@ -117,29 +115,29 @@ func Run(namespace, k8sContext, devPath, language, workDir string, overwrite boo
 
 	dev.Context = k8sContext
 
-	if checkForDeployment {
-		r, container, err := getResource(ctx, namespace, k8sContext)
+	if checkForRunningApp {
+		app, container, err := getRunningApp(ctx, namespace, k8sContext)
 		if err != nil {
 			return err
 		}
-		if r == nil {
+		if app == nil {
 			dev.Autocreate = true
 			linguist.SetForwardDefaults(dev, language)
 		} else {
 			dev.Container = container
 			if container == "" {
-				container = r.PodTemplateSpec.Spec.Containers[0].Name
+				container = app.PodSpec().Containers[0].Name
 			}
 
-			suffix := fmt.Sprintf("Analyzing %s '%s'...", strings.ToLower(string(r.ObjectType)), r.Name)
+			suffix := fmt.Sprintf("Analyzing %s '%s'...", app.Kind(), app.Name())
 			spinner := utils.NewSpinner(suffix)
 			spinner.Start()
-			err = initCMD.SetDevDefaultsFromResource(ctx, dev, r, container, language)
+			err = initCMD.SetDevDefaultsFromApp(ctx, dev, app, container, language)
```

Comment:
At this point you may use another smaller interface as the parameter to this function to something that simply provides labels and annotations. I believe doing it this way will make things a little simpler at the start and possibly allow for some cleaner factoring. Also, in the long run will provide a stronger abstraction allowing for further cleanup when/if needed.